src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java

```diff
@@ -135,8 +135,10 @@ public FileUploadSubmission save(FileUploadSubmission fileUploadSubmission, Mult
             }
         }
         // update submission properties
-        // NOTE: from now on we always set submitted to true to prevent problems here!
-        fileUploadSubmission.setSubmitted(true);
+        // NOTE: from now on we always set submitted to true to prevent problems here! Except for late submissions of course exercises to prevent issues in auto-save
```

Comment:
The commit introducing this comment says `fixes the bug that auto-save overrides previous submissions to submitted = false` (which is way more informative than `from now on we always set submitted to true to prevent problems here` fwiw). Is there a way to do a late submission to an exercise you already submitted to? I didn't find one for text or modeling exercises. (=> If that was possible it would break here likely.)